examples/go-libp2p-server/main.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/libp2p/go-libp2p"
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/network"
+	"github.com/libp2p/go-libp2p/core/peer"
+	webrtc "github.com/libp2p/go-libp2p/p2p/transport/webrtc"
+)
+
+var listenerIp = net.IPv4(127, 0, 0, 1)
+
+func init() {
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return
+	}
+	for _, iface := range ifaces {
+		if iface.Flags&net.FlagUp == 0 {
+			continue
+		}
+		addrs, err := iface.Addrs()
+		if err != nil {
+			return
+		}
+		for _, addr := range addrs {
+			// bind to private non-loopback ip
+			if ipnet, ok := addr.(*net.IPNet); ok && !ipnet.IP.IsLoopback() && ipnet.IP.IsPrivate() {
+				if ipnet.IP.To4() != nil {
+					listenerIp = ipnet.IP.To4()
+					return
+				}
+			}
+		}
+	}
+}
+
+func echoHandler(stream network.Stream) {
+	for {
+		reader := bufio.NewReader(stream)
+		str, err := reader.ReadString('\n')
+		log.Printf("err: %s", err)
+		if err != nil {
+			return
+		}
+		log.Printf("echo: %s", str)
+		_, err = stream.Write([]byte(str))
+		log.Printf("err: %s", err)
+		if err != nil {
+			return
+		}
+
+	}
+}
+
+func main() {
+	host := createHost()
+	host.SetStreamHandler("/echo/1.0.0", echoHandler)
+	defer host.Close()
+	remoteInfo := peer.AddrInfo{
+		ID:    host.ID(),
+		Addrs: host.Network().ListenAddresses(),
+	}
+
+	remoteAddrs, _ := peer.AddrInfoToP2pAddrs(&remoteInfo)
```

Comment:
you know a language has poor error handling when examples usually omit the error handling  :D